xbmc/video/guilib/VideoVersionHelper.cpp

```diff
@@ -43,33 +45,38 @@ std::shared_ptr<CFileItem> CVideoActionProcessorHelper::ChooseVideoVersion()
           else
           {
             CFileItem defaultVersion;
-            db.GetDefaultVideoVersion(m_item->GetVideoContentType(),
-                                      m_item->GetVideoInfoTag()->m_iDbId, defaultVersion);
+            db.GetDefaultVideoVersion(item->GetVideoContentType(), item->GetVideoInfoTag()->m_iDbId,
```

Comment:
Information related to PR#24148 please, and not a blocker for this PR: when is the information of m_item (now item) not good and needs to retrieve default version from db?
If it wasn't needed anymore, together with your other change merged yesterday to remove the default version switch in db, there wouldn't be any need for a `GetDefaultVideoVersion()` anymore, simplifying 24148.

Reply:
@CrystalP not sure whether this actually is dead code (now), tbh. Could be, but if it was, as this is the only place in our code where the value of the setting `SETTING_MYVIDEOS_SELECTDEFAULTVERSION` is checked, this would mean that the setting can be removed. This would be great, but I'm uncertain whether I understood all this correct, I'm hesitant to remove this code.

Reply:
Found how to get there,I wouldn't remove just yet. It's the same answer as in #24148, `GetDefaultVideoVersion()` prepares an info tag/file item with some fields altered from default. That's a recurring theme, maybe the code creating those videoversion infotags (in  `CVideoDatabase::GetVideoVersions()` ) belongs in a common path of item creation and some exceptions could be removed.

Reply:
> Found how to get there

@CrystalP can you please share how, so I can have a look later?

Reply:
Found it. Yes, we need to keep this (for now).